### PR TITLE
refactor: bump to v2.2

### PR DESCRIPTION
**Pull request by iain-anderson**
Signed-off-by: Iain Anderson <iain@iotechsys.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-coap-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by iain-anderson** [comment]
Also update to Alpine 3.16, which includes its own libcbor, so no need to build that anymore

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+
+<a name="2.2.0"></a>
+
+## v2.2.0 - 2022-08-08
+
+### Features ✨
+- Bump to SDK v2.2 ([#41](https://github.com/edgexfoundry/device-coap-c/issues/41))
+- Upgrade Alpine 3.16 ([#40](https://github.com/edgexfoundry/device-coap-c/issues/40))
+- Use secret store for PSK key ([#33](https://github.com/edgexfoundry/device-coap-c/issues/33)) ([#34](https://github.com/edgexfoundry/device-coap-c/issues/34)) ([#402b1dc](https://github.com/edgexfoundry/device-coap-c/commits/402b1dc))
+- Bump to SDK v2.1 ([#30](https://github.com/edgexfoundry/device-coap-c/issues/30)) ([#12271f6](https://github.com/edgexfoundry/device-coap-c/commits/12271f6))
```
Comment (lenny-intel) [inline CHANGELOG.md:13]: This should be under v2.1.0 section

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by iain-anderson** [comment]
Fixed changelog and added missing default config values

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**